src/dds_service.py

```diff
@@ -72,3 +74,315 @@ def builtin_observer(domain_id, dds_data, running):
                 dds_data.remove_endpoint(domain_id, sub)
 
     logging.info(f"builtin_observer({domain_id}) ... DONE")
+
+
+class dds_durability_kind(OrderedEnum):
+    DDS_DURABILITY_VOLATILE = 0
+    DDS_DURABILITY_TRANSIENT_LOCAL = 1
+    DDS_DURABILITY_TRANSIENT = 2
+    DDS_DURABILITY_PERSISTENT = 3
+
+class dds_history(OrderedEnum):
+    DDS_HISTORY_KEEP_LAST = 0
+    DDS_HISTORY_KEEP_ALL = 1
+
+class dds_ownership(OrderedEnum):
+    DDS_OWNERSHIP_SHARED = 0
+    DDS_OWNERSHIP_EXCLUSIVE = 1
+
+class dds_liveliness(OrderedEnum):
+    DDS_LIVELINESS_AUTOMATIC = 0
+    DDS_LIVELINESS_MANUAL_BY_PARTICIPANT = 1
+    DDS_LIVELINESS_MANUAL_BY_TOPIC = 2
+
+class dds_reliability(OrderedEnum):
+    DDS_RELIABILITY_BEST_EFFORT = 0
+    DDS_RELIABILITY_RELIABLE = 1
+
+class dds_destination_order(OrderedEnum):
+    DDS_DESTINATIONORDER_BY_RECEPTION_TIMESTAMP = 0
+    DDS_DESTINATIONORDER_BY_SOURCE_TIMESTAMP = 1
+
+class dds_presentation_access_scope(OrderedEnum):
+    DDS_PRESENTATION_INSTANCE = 0
+    DDS_PRESENTATION_TOPIC = 1
+    DDS_PRESENTATION_GROUP = 2
+
+class dds_ignorelocal(OrderedEnum):
+    DDS_IGNORELOCAL_NONE = 0
+    DDS_IGNORELOCAL_PARTICIPANT = 1
+    DDS_IGNORELOCAL_PROCESS = 2
+
+class dds_type_consistency(OrderedEnum):
+    DDS_TYPE_CONSISTENCY_DISALLOW_TYPE_COERCION = 0
+    DDS_TYPE_CONSISTENCY_ALLOW_TYPE_COERCION = 1
+
+class dds_qos_policy_id(Enum):
```

Comment:
These `POLICY_ID` definitions seem out-of-place here. It would make more sense for `qos.py` in the python binding to define them as part of the class definitions. Similarly, it would make more sense for the `OrderedEnum`s defined above to be used by `qos.py` to do the conversion between Python and C. (It currently uses the numbers, not symbolic constants.)

The advantage of defining everything here is of course not having to change the binding for the tool, but in this case, I think it simply shows that the binding took a shortcut too many.

Reply:
Let's keep it for now, add it to the Python binding later. Keeping it here for now means you have more freedom and can move faster 😀